mne/epochs.py

```diff
@@ -1342,6 +1342,8 @@ def _get_data(self, out=True, verbose=None):
                         epoch = self._data[idx]
                 else:  # from disk
                     epoch_noproj = self._get_epoch_from_raw(idx)
+                    if epoch_noproj is None:  # Dropped due to bad segment
+                        continue
```

Comment:
It is to make message on ``drop_log`` clearer. ``_is_good_epoch`` later adds ``NO_DATA`` to ``drop_log`` if data is None. So in case of bad segment, the log would have two entries without this.